skfem/element/discrete_field.py

```diff
@@ -52,28 +52,31 @@ def __array_wrap__(self, out_arr, context=None):
         # attributes are invalidated after ufuncs
         return np.array(out_arr)
 
+    def get(self, n):
+        if n == 0:
+            return np.array(self)
+        return getattr(self, self._extra_attrs[n - 1])
+
     @property
     def astuple(self):
         return tuple(self.get(i) for i in range(len(self._extra_attrs) + 1))
 
     @property
     def value(self):
+        # increase backwards-compatibility
         return self
```

Comment:
Is this line right? It allows `x.value.grad`. Previously `.value` was a naked ndarray. This is subtle and the line might fine. I wanted to point out the change from previous behavior since this is here for backward comapt.

Reply:
Alright, good point. This is the reason why w.fun.grad now also works. Test suite passes, no changes related to this required. Since we are already breaking the backwards compatibility to some extent we could do it here also. I need to think about it.

Reply:
One option is to cast to ndarray here and then in FormParams not use DiscreteField.value.

Reply:
Can you go so far as to say that `.value` is deprecated? I'm not sure there is a use case for it now, other than supporting previously written code?

Reply:
Fair enough, should get rid of `.value` from all code under `skfem`. I have the tendency of leaving around deprecated stuff so that I remember the state of affairs and support those as long as it's reasonable.